### ISSUE TITLE
Upgrade of instance_template and compute_instance from v10.1.1 to v11.1.0 triggers replacement of the VM

### ISSUE THREAD

**Issue by drengskapr** [terraform-google-modules/terraform-google-vm#402]
### TL;DR

Hello there! My VM was created using `instance_template` and `compute_instance` modules both v10.1.1. Now after I have upgraded these modules version to v11.1.0 `terraform plan` triggers replacement of that VM. Nothing apart from module versions is changed.

### Expected behavior

`terraform plan` must not trigger replacement of the VM

### Observed behavior

```
source_instance_template  = "https://www.googleapis.com/compute/v1/projects/my-project/global/instanceTemplates/vm-xxx" -> "https://www.googleapis.com/compute/beta/projects/my-project/global/instanceTemplates/vm-xxx" # forces replacement
```

Since v11.0.0 `instance_template` module uses `google-beta` provider instead of `google`, thus the `source_instance_template` is changed. Note `https://www.googleapis.com/compute/v1` is changed to `https://www.googleapis.com/compute/beta`.

### Terraform Configuration

```hcl
module "instance_template" {
  source               = "terraform-google-modules/vm/google//modules/instance_template"
  version              = "11.1.0"
  name_prefix          = "vm-vpn-template"
  metadata             = var.vm_vpn_metadata
  tags                 = var.vm_vpn_tags
  region               = var.default_region
  project_id           = var.default_project
  subnetwork           = var.management_subnetwork_name
  service_account      = var.service_account
  machine_type         = var.vm_vpn_machine_type
  disk_type            = var.vm_vpn_disk_type
  disk_size_gb         = var.vm_vpn_disk_size_gb
  source_image_family  = var.vm_vpn_source_image_family
  source_image_project = var.vm_vpn_source_image_project
  can_ip_forward       = false
  enable_shielded_vm   = true

  shielded_instance_config = {
    enable_secure_boot          = true
    enable_vtpm                 = false
    enable_integrity_monitoring = false
  }

}

module "compute_instance" {
  source              = "terraform-google-modules/vm/google//modules/compute_instance"
  version             = "11.1.0"
  region              = var.default_region
  zone                = var.default_zone
  subnetwork          = var.management_subnetwork_name
  num_instances       = var.vm_vpn_num_instances
  hostname            = "vpn"
  instance_template   = module.instance_template.self_link

  access_config = [{
    nat_ip       = google_compute_address.vpn_external_ip.address
    network_tier = var.vm_vpn_network_tier
  }, ]
}
```

### Terraform Version

```sh
v1.7.5
```

### Additional information

If I use `instance_template` v10.1.1 and `compute_instance` v11.1.0 everything is working fine.
